solr/core/src/java/org/apache/solr/search/neural/KnnQParser.java

```diff
@@ -80,44 +82,68 @@ public Query parse() {
     float[] parsedVectorToSearch = parseVector(vectorToSearch, denseVectorType.getDimension());
 
     return denseVectorType.getKnnVectorQuery(
-        schemaField.getName(), parsedVectorToSearch, topK, getFilterQuery());
+        schemaField.getName(), parsedVectorToSearch, topK, buildPreFilter());
   }
 
-  private Query getFilterQuery() throws SolrException {
-    if (!isFilter()) {
+  private Query buildPreFilter() throws SolrException {
```

Comment:
IMO its name was just fine.  "pre-filter" is not something I've ever heard someone say in Solr.